isobus/src/can_network_manager.cpp

```diff
@@ -557,15 +558,14 @@ namespace isobus
 			if (nullptr == foundControlFunction)
 			{
 				// If we still haven't found it, it might be a partner. Check the list of partners.
-				for (std::size_t i = 0; i < PartneredControlFunction::partneredControlFunctionList.size(); i++)
+				for (const auto &partner : partneredControlFunctions)
 				{
-					if ((nullptr != PartneredControlFunction::partneredControlFunctionList[i]) &&
-					    (PartneredControlFunction::partneredControlFunctionList[i]->get_can_port() == rxFrame.channel) &&
-					    (PartneredControlFunction::partneredControlFunctionList[i]->check_matches_name(NAME(claimedNAME))))
+					if ((partner->get_can_port() == rxFrame.channel) &&
+					    (partner->check_matches_name(NAME(claimedNAME))))
 					{
-						PartneredControlFunction::partneredControlFunctionList[i]->address = claimedAddress;
-						PartneredControlFunction::partneredControlFunctionList[i]->controlFunctionNAME = NAME(claimedNAME);
-						foundControlFunction = std::shared_ptr<ControlFunction>(PartneredControlFunction::partneredControlFunctionList[i]);
+						partner->address = claimedAddress;
+						partner->controlFunctionNAME = NAME(claimedNAME);
+						foundControlFunction = partner;
```

Comment:
At some point I think we lost the logging message "A partner with name X has claimed address Y on channel Z" which might be nice to have when reading the debug logs so that users can tell by its absence if their NAME filter's didn't match any ECUs? Or maybe it might make sense to check the list of partners at a low frequency and emit a message if their address is invalid? Open to thoughts on that.

Reply:
> Or maybe it might make sense to check the list of partners at a low frequency and emit a message if their address is invalid? 

I'd say that is up to the user to find out. They should be able to make such a function really easy themselves when we expose the list of partners.

> At some point I think we lost the logging message "A partner with name X has claimed address Y on channel Z" which might be nice to have when reading the debug logs so that users can tell by its absence if their NAME filter's didn't match any ECUs?

Hmm, it seems to be at debug level now: 
https://github.com/Open-Agriculture/AgIsoStack-plus-plus/blob/72a069bf0d7f2584c00f53220768cd1fe3459461/isobus/src/can_network_manager.cpp#L660
It might make sense to bump it to the info level instead, and then use a message along the lines of "A partner with name X has claimed address Y on channel Z" yeah.